src/test/java/com/marek/onlinebookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,179 @@
+import javax.sql.DataSource;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ShoppingCartControllerTest {
+    public static final int QUANTITY = 1;
+    private static final String EMAIL = "okok@email.com";
+    private static final String PASSWORD =
+            "$2a$10$EhBFr.PagMjT0P0EYqRL/.KjPUA2vRSutGZo92Xr9Hh/JwwAJq/vi";//password
+    private static final String FIRST_NAME = "First Name";
+    private static final String LAST_NAME = "Last Name";
+    private static final String SHIPPING_ADDRESS = "address";
+    private static final Set<Role> EMPTY_ROLES = new HashSet<>();
+    private static final String TITLE = "Title";
+    private static final String AUTHOR = "author";
+    private static final String ISBN = "123123";
+    private static final BigDecimal PRICE = BigDecimal.valueOf(12.12);
+    private static final String DESCRIPTION = "image";
+    private static final String COVER_IMAGE = "description";
+    private static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("db/cart/controller/delete-carts.sql")
+            );
+        }
+    }
+
+    @Test
+    @DisplayName("""
+            Update item in cart with valid input returns CartItemDto
+            """)
+    @WithMockUser(username = "user", roles = "USER")
+    @SneakyThrows
+    @Sql("classpath:db/cart/controller/add-data-for-cart-test.sql")
```

Comment:
Add `executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD`.
You should also undo changes that this operation may have made to the database. For example, you can do it this way:

```java
    @Sql(scripts = {"classpath:database/books/add-three-books-with-categories.sql"},
            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
    @Sql(scripts = {"classpath:database/shoppingcart/delete_shopping_cart.sql"},
            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
```